### DISCUSSION TITLE
Snags on Windows 10

### DISCUSSION THREAD

**Discussion by mikelevins** [rabbibotton/clog#83] [General]
I was working on some infrastructure that turns out to more or less duplicate clog's features. I would be glad to abandon my work and use clog instead, but ran into some snags using it on Windows.

I tried the latesst version that's in QuickLisp. In order to get it to work I had to:

1. put sqlite3.dll in the same directory as sbcl.exe (there are actually other ways, but they are more complicated; I can modify my PATH or I can use a locally-modified version of cl-sqlite to help it find the DLL).
2. push :HUNCHENTOOT-NO-SSL onto *features* (or figure out where to get a version of libcrypto that I can use)
3. push :WEBSOCKET-DRIVER-NO-SSL onto *features* (or figure out where to get a version of libcrypto that I can use)

Once I did these things I could run the demos and tutorials, but they always signal the following error:

Subprocess with command "explorer \"http://127.0.0.1:8080\""
 exited with error code 1
   [Condition of type UIOP/RUN-PROGRAM:SUBPROCESS-ERROR]

Restarts:
 0: [CONTINUE] IGNORE-ERROR-STATUS
 1: [*ABORT] Return to SLIME's top level.
 2: [ABORT] abort thread (#<THREAD "repl-thread" RUNNING {1005649E53}>)

Backtrace:
  0: (UIOP/RUN-PROGRAM::%CHECK-RESULT 1 :COMMAND "explorer \"http://127.0.0.1:8080\"" :PROCESS NIL :IGNORE-ERROR-STATUS NIL)
  1: (UIOP/RUN-PROGRAM::%USE-SYSTEM "explorer \"http://127.0.0.1:8080\"")
  2: (SB-INT:SIMPLE-EVAL-IN-LEXENV (CLOG:RUN-DEMO 3) #<NULL-LEXENV>)
  3: (EVAL (CLOG:RUN-DEMO 3))

Finally, I tried again with the head of the github repo, but immediately ran into:

COMPILE-FILE-ERROR while
compiling #<CL-SOURCE-FILE "clog" "clog-gui">
   [Condition of type UIOP/LISP-BUILD:COMPILE-FILE-ERROR]

Restarts:
 0: [RETRY] Retry compiling #<CL-SOURCE-FILE "clog" "clog-gui">.
 1: [ACCEPT] Continue, treating compiling #<CL-SOURCE-FILE "clog" "clog-gui"> as having been successful.
 2: [RETRY] Retry ASDF operation.
 3: [CLEAR-CONFIGURATION-AND-RETRY] Retry ASDF operation after resetting the configuration.
 4: [RETRY] Retry ASDF operation.
 5: [CLEAR-CONFIGURATION-AND-RETRY] Retry ASDF operation after resetting the configuration.
 --more--

Backtrace:
  0: (UIOP/LISP-BUILD:CHECK-LISP-COMPILE-RESULTS NIL T T "~/asdf-action::format-action/" ((#<ASDF/LISP-ACTION:COMPILE-OP > . #<ASDF/LISP-ACTION:CL-SOURCE-FILE "clog" "clog-gui">)))
  1: ((SB-PCL::EMF ASDF/ACTION:PERFORM) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:COMPILE-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "clog" "clog-gui">)
  2: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
  3: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS :AROUND (T T)) #<ASDF/LISP-ACTION:COMPILE-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "clog" "clog-gui">) [fast-method]
  4: ((:METHOD ASDF/PLAN:PERFORM-PLAN (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {100E82D4A3}>) [fast-method]
  5: ((FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))

I'm willing to believe that I'm doing something foolish that is causing these errors, and would be glad of correction. If I can use Clog on Windows, Linux, and macOS, it looks like it'll save me a lot of work building similar infrastructure myself.

**Comment by rabbibotton** [comment]
Can you describe you lisp environment? For testing on windows I use portacle as mentioned in the readme.

**Reply by mikelevins** [reply]
I'm not using portacle.

I'm using sbcl 2.1.9 at C:\Program Files\Stell Bank Common Lisp\sbcl.exe with Emacs 27.2 from https://www.gnu.org/software/emacs/download.html, and the latest quicklisp.

I'm using use SLIME 2.26.1 insstalled using quicklisp.

I thknk that's about all there is to it.

**Reply by mikelevins** [reply]
Well, except for the typo where I wrote "Stell" instead of "Steel" >.>

**Comment by rabbibotton** [comment]
so the Subprocess with command "explorer "http://127.0.0.1:8080\"" occurs
in windows 11 (maybe 10 too in more up to date feature packs) and is a
problem with trivial browser open, seems the command that has worked for
years to open a url on windows returns an error. I use it as a convenience.
For now I would ignore error and just open up that url until someone fixes
trivial browser open.

I would say uninstall clog.Then use git clone clog into a directory that is
asdf friendly like your home directory /common-lisp/clog and try again.

**Reply by mikelevins** [reply]
I can most likely write my own substitute for trivial-browser-open (or repurpose the one I already wrote for my own project).

I already tried your second suggestion; that's how I tried the head of the github repo. That produced the second error trace I posted.

**Comment by rabbibotton** [comment]
I’ll see if I can recreate but setting up an environment similar to yours.

**Comment by mikelevins** [comment]
Thanks. Even if you can’t reproduce it, the different results and any differences in environments may offer me clues.

**Comment by rabbibotton** [comment]
True however the error is a configuration issue not code but agreed will
lead to better support.

**Reply by mikelevins** [reply]
For what it's worth, I seem to have arrived at a working install.

I deleted my quicklisp install and resinstalled fresh (again). I also deleted my ASDF cache and the clone of the head of the clog repo in ~/common-lisp (or, more accurately, c:\Users\mikel\common-lisp).

The asd file for my project pushes :HUNCHENTOOT-NO-SSL and :WEBSOCKET-DRIVER-NO-SSL on \*features\* before defining the system.

sqlite3.dll is in the same directory as sbcl.exe.

I added this to the app's code:

#+os-windows
(eval-when (:compile-toplevel :load-toplevel :execute)
  (setf trivial-open-browser::+format-string+ "start msedge ~S"))

With these changes, clog appears at first glance to start on Windows without problems, and I'll proceed with my planned experiments.

Thank you for your kind attention.

**Comment by rabbibotton** [comment]
Fantastic
